### ISSUE TITLE
inited Helm config failed

### ISSUE THREAD

**Issue by anhquan075** [iter8-tools/iter8#1447]
**Describe the bug**
I try to run iter8 for the first time. But in the Helm installation step, I got the error below:
```
INFO[2023-04-19 17:42:01] inited Helm config                           
ERRO[2023-04-19 17:42:05] experiment launch failed                      stack-trace=below ... 
::Trace:: unable to build kubernetes objects from release manifest: error validating "": error validating data: [ValidationError(Job.spec.template.spec): unknown field "resources" in io.k8s.api.core.v1.PodSpec, ValidationError(Job.spec.template.spec.securityContext): unknown field "readOnlyRootFilesystem" in io.k8s.api.core.v1.PodSecurityContext]
Error: experiment launch failed
```
**To Reproduce**
Run the [tutorial](https://github.com/kserve/kserve/tree/master/docs/samples/performance-testing) of Kserve for performance testing

**Screenshots**
<img width="1323" alt="image" src="https://user-images.githubusercontent.com/54442768/233052088-2992a81b-84f0-4a54-83b9-8f946e8ded48.png">

**Desktop (please complete the following information):**
 - OS: MacOS 12.6
 - iter8 version: 13
 - Kubernetes version: 1.22.2

**Comment by kalantar** [comment]
were you following one of the tutorials or trying something else? Can you provide the full command you tried?

**Comment by sriumcp** [comment]
@Alan-Cha FYI.

Perhaps, the `iter8` chart needs to be reverted back? It may be that the `kubescape` linting related changes to `iter8` has broken something?

**Comment by anhquan075** [comment]
> @kalantar were you following one of the tutorials or trying something else? Can you provide the full command you tried?

I just follow the example in [Kserve tutorial](https://github.com/kserve/kserve/tree/master/docs/samples/performance-testing) for the first time. My full command is kind of same as [this](https://github.com/kserve/kserve/tree/master/docs/samples/performance-testing), just change the endpoint and payload url to fit with my experiment: 
```
iter8 k launch \
--set "tasks={ready,http,assess}" \
--set ready.isvc=chiller-isvc \
--set ready.timeout=600s \
--set http.url=http://modelmesh-serving.modelmesh-serving.svc.cluster.local/v2/models/chiller-isvc/infer \
--set http.payloadURL=<my-payload-link.json> \
--set http.contentType="application/json" \
--set http.warmupNumRequests=50 \
--set assess.SLOs.upper.http/latency-mean=500 \
--set assess.SLOs.upper.http/latency-p90=1000 \
--set assess.SLOs.upper.http/error-count=0 \
--set runner=job
```

**Comment by sriumcp** [comment]
This seems like a regression caused by the recent Kubescape based linting.

***

Resources appear to be set in the [wrong place](https://github.com/iter8-tools/hub/blob/94365c4861ac0962304b9df01b7533946e8ec0ed/charts/iter8/templates/_k-job.tpl#L28). They should be set [per-container as in this example](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#example-1).

Same is the case `readOnlyRootFileSystem`. Security context for pod and container are different objects. API reference is  [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#securitycontext-v1-core). Security context for pod is through the `podSecurityContext` object, while for a container, it is through the `securityContext` object.

***

We will want to add CI to the `hub` repo to catch this in a test as part of the fix. We should also add tests for the other charts (at the minimum, the tests should check for a successful Helm install).

**Comment by Alan-Cha** [comment]
@anhquan075 I believe that this issue has been resolved with the latest version of Iter8 [v0.13.14](https://github.com/iter8-tools/iter8/releases/tag/v0.13.14). I tested the [Kserve tutorial](https://github.com/kserve/kserve/tree/master/docs/samples/performance-testing) on Kubernetes v1.22.7 without any issue. Please try again and let me know if there are any probelms.

For now, I will close the issue and I will reopen it you have any other concerns.
